### PR TITLE
refactor(structured-list): remove <p> and content class

### PR DESCRIPTION
**Pull request by hellobrian**
## Overview

While porting Structured List to React, realized that I didn't need to use `<p>` or `.bx--structured-list-content` class.

### CODE REVIEW

File: src/components/structured-list/_structured-list.scss [hunk 1]
```diff
@@ -87,14 +87,17 @@
...
     @include padding-td;
     position: relative;
     display: table-cell;
-    font-size: rem(14px);
   }
 
+  // Deprecated class
   .bx--structured-list-content {
-    font-size: rem(14px);
+    @include font-size('14');
```
Comment (hellobrian) [inline src/components/structured-list/_structured-list.scss:100]: Keeping the class selector to avoid breaking change.

**Comment by chrisdhanaraj** [comment]
Just to confirm - is the idea that we will _never_ want people to add more content into the structured list other than the current styles? 👍 ✅ if so

**Comment by marijohannessen** [comment]
👍 ✅

**Comment by hellobrian** [comment]
> is the idea that we will never want people to add more content into the structured list other than the current styles? - @chrisdhanaraj 

So far, we only expect text content to be added to structured-list.
With that in mind, the user should be able to directly add their content into `div.bx--structured-list-td` and get the correct styles: 

image

Previously, `p.bx--structured-list-content` was also doing this but it was redundant since `div.bx--structured-list-td` can do this as well. One less class for the user to worry about and makes it more flexible for users to pass in elements that are not `<p>`.

**Pull request merged**